framework/src/actions/CreateDisplacedProblemAction.C

```diff
@@ -26,23 +26,35 @@ InputParameters validParams<InitDisplacedProblemAction>()
   return params;
 }
 
-InitDisplacedProblemAction::InitDisplacedProblemAction(InputParameters parameters) :
+CreateDisplacedProblemAction::CreateDisplacedProblemAction(InputParameters parameters) :
     Action(parameters)
 {
 }
 
-InitDisplacedProblemAction::~InitDisplacedProblemAction()
+CreateDisplacedProblemAction::~CreateDisplacedProblemAction()
 {
 }
 
 void
-InitDisplacedProblemAction::act()
+CreateDisplacedProblemAction::act()
 {
-  if (isParamValid("displacements"))
+  if (isParamValid("displacements") && _displaced_mesh.get() != NULL)
```

Comment:
I was just making it consistent with CreateProblemAction, I didn't think about it.